### PR TITLE
[WIP] Type export experimentation

### PR DESCRIPTION
**Pull request by wagenet**
Includes #20090

### CODE REVIEW

**Review by chriskrycho — CHANGES_REQUESTED** [review]
Couple of nits from the modules affected, will deeply-read-and-understand `build-docs.js` next.

File: packages/@ember/-internals/runtime/lib/system/core_object.d.ts [hunk 1]
```diff
@@ -14,7 +14,6 @@ import { Owner } from '@ember/-internals/owner';
 /** @internal */
 export type EmberClassConstructor<T> = new (owner: Owner) => T;
 
-/** @internal */
```
[the remark this answers is not in the document]
Comment (wagenet) [reply packages/@ember/-internals/runtime/lib/system/core_object.d.ts]: Good question. I'll need to look into it more.

**Comment by locks** [comment]
I see a lot of conflicts. Has the work been done elsewhere?

**Comment by kategengler** [comment]
Is this still relevant? It is a draft PR

**Comment by chriskrycho** [comment]
I *believe* this was superceded in its entirety by #20449 (as well as the foundational PRs which preceded it); this was Peter's admirable first pass at trying to get types publishing, and was invaluable prior art in informing the Ember TypeScript team that we were going to have to do the *absolute shenanigans* that we did in #20449 to get things published. Closing accordingly!

**Pull request closed without merging**